File: server/echoes-of-the-metro/src/main/java/com/kevindonovan/eotm/echoes_of_the_metro/models/DTOs/EventResponse.java
package com.kevindonovan.eotm.echoes_of_the_metro.models.DTOs;

import com.kevindonovan.eotm.echoes_of_the_metro.models.Badge;
import com.kevindonovan.eotm.echoes_of_the_metro.models.EventType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class EventResponse {

    private int eventId;
    private String text;
    private EventType eventType;
    private int scrapFound;
    private int fuelFound;
    private int mgrCollected;
    private int locationId;
    private Badge badge;
    private String soundPath;
    private String mediaPath;
}

File: client/eotm-client/src/types/response/EventResponse.ts
import type { EventType } from "../enums/EventType";
import type { BadgeResponse } from "./BadgeResponse";

export type EventResponse = {
    eventId: number;
    text: string;
    eventType: EventType;
    scrapFound: number;
    fuelFound: number;
    mgrCollected: number;
    locationId: number;
    badge: BadgeResponse | null;
    soundPath: string;
    mediaPath: string;
}